libraries/hotstuff/qc_chain.cpp

```diff
@@ -235,12 +229,20 @@ namespace eosio::hotstuff {
    }
 
 
-   qc_chain::qc_chain(name id, base_pacemaker* pacemaker, std::set<name> my_producers, fc::logger& logger)
-      : _pacemaker(pacemaker),
+   qc_chain::qc_chain(name id, base_pacemaker* pacemaker, std::set<name> my_producers, fc::logger& logger, std::string safety_state_file)
+      : _id(id),
+        _pacemaker(pacemaker),
         _my_producers(std::move(my_producers)),
-        _id(id),
         _logger(logger)
    {
+
+      //todo : read liveness state / select initialization heuristics ?
+
+      if (safety_state_file!=std::string() && std::filesystem::exists(safety_state_file)){
```

Comment:
`!safety_state_file.empty()` instead of `!=std::string()`

Reply:
I'm using the empty `_safety_state_file` string member in `qc_chain` to signal turning off the safety state file feature altogether.